metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran.py

```diff
@@ -232,16 +233,21 @@ def _generate_datajob_from_connector(self, connector: Connector) -> DataJob:
             "sync_frequency": str(connector.sync_frequency),
             "destination_id": connector.destination_id,
         }
-        datajob.properties = {
-            **connector_properties,
-            **lineage_properties,
-        }
+
+        datajob.set_custom_properties({**connector_properties, **lineage_properties})
 
         return datajob
 
     def _generate_dpi_from_job(self, job: Job, datajob: DataJob) -> DataProcessInstance:
+        # hack: convert to old instance for DataProcessInstance.from_datajob compatibility
+        datajob_v1 = DataJobV1(
+            id=job.job_id,
+            flow_urn=datajob.flow_urn,
+            platform_instance=self.config.platform_instance,
+            name=datajob.name,
+        )
         return DataProcessInstance.from_datajob(
```

Comment:
eh this is very hacky and i don't like this solution, but seems like we need to do a lot of job with `api.entities.dataprocessinstance` code to be compatible with the SDK v2. i'd rather just add DPI to SDK v2 rather than trying to make this compatible.